exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -69,6 +69,11 @@ type AzureManagedMachinePoolSpec struct {
 	// MaxPods specifies the kubelet --max-pods configuration for the node pool.
 	// +optional
 	MaxPods *int32 `json:"maxPods,omitempty"`
+
+	// OsDiskType specifies the OS disk type for each node in the pool. Allowed values are 'Ephemeral' and 'Managed'.
+	// +kubebuilder:validation:Enum=Ephemeral; Managed
+	// +optional
+	OsDiskType string `json:"osDiskType,omitempty"`
```

Comment:
should we use a pointer here to make it truly optional? 
and if it's not optional in sense, there should be always some default value can we add this to mutation webhook?

Reply:
+1 

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
done